src/Http/Http.Extensions/gen/StaticRouteHandlerModel/Emitters/EndpointParameterEmitter.cs

```diff
@@ -87,15 +153,31 @@ internal static string EmitServiceParameterPreparation(this EndpointParameter en
                         {endpointParameter.EmitParameterDiagnosticComment()}
 """);
 
+        var assigningCode = endpointParameter.IsOptional ?
+            $"httpContext.RequestServices.GetService<{endpointParameter.Type}>();" :
+            $"httpContext.RequestServices.GetRequiredService<{endpointParameter.Type}>()";
+
         // Requiredness checks for services are handled by the distinction
         // between GetRequiredService and GetService in the AssigningCode.
```

Comment:
```suggestion
        // between GetRequiredService and GetService in the assigningCode.
```

Actually, is this comment even necessary anymore?

Reply:
> Actually, is this comment even necessary anymore?

Ish. It would be helpful to clarify here that requiredness checks for services will result in an exception being thrown (by virtue of `GetRequiredService` being called) not the 400 status code behavior that exists for other types.